x-pack/auditbeat/module/system/process/process.go

```diff
@@ -84,12 +94,21 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 	ms.config = defaultConfig
 	ms.log = logp.NewLogger(metricsetName)
+	ms.SystemMetricSet = system.NewSystemMetricSet(base)
 
 	if err := base.Module().UnpackConfig(&ms.config); err != nil {
 		return nil, fmt.Errorf("failed to unpack the %v/%v config: %w", system.ModuleName, metricsetName, err)
 	}
 
-	return NewFromSysInfo(base, ms)
+	if runtime.GOOS == "linux" && ms.config.Backend == "kernel_tracing" {
+		if qm, err := NewFromQuark(ms); err == nil {
+			return qm, nil
+		} else {
+			ms.log.Errorf("can't use kernel_tracing, falling back to procfs: %w", err)
```

Comment:
You cannot use `%w` with `logp.Errorf`. That's only for wrapping errors with `fmt.Errorf`.

Switch to `%v` or change to a structured log message like `ms.log.Errorw("my static message", "error", err)` (which will render as a JSON object containing the `err.Error()` value in the ECS `error.message` field).

Reply:
well spotted, many thanks!